scripts/safe_template_linter.py

```diff
@@ -553,27 +544,39 @@ def prepare_results(self, full_path, string_lines):
             self.lines.append(string_lines.line_number_to_line(line_number))
         self._mark_expression_disabled(string_lines)
 
-    def print_results(self, out):
+    def print_results(self, options, out):
         """
         Prints the results represented by this rule violation.
 
         Arguments:
+            options: A list of the following options:
+                list_files: True to print only file names, and False to print
+                    all violations.
+                verbose: True for multiple lines of context, False single line.
             out: output file
 
         """
-        for line_number in range(self.start_line, self.end_line + 1):
+        if options['verbose']:
+            end_line = self.end_line + 1
+        else:
+            end_line = self.start_line + 1
+        for line_number in range(self.start_line, end_line):
             if line_number == self.start_line:
                 column = self.start_column
                 rule_id = self.rule.rule_id + ":"
             else:
                 column = 1
                 rule_id = " " * (len(self.rule.rule_id) + 1)
+            try:
+                line = self.lines[line_number - self.start_line].encode(encoding='utf-8')
+            except (TypeError, IndexError):
+                line = "Error printing line with line_number={} and start_line={}.".format(line_number, self.start_line)
```

Comment:
@nedbat: A programming error elsewhere along the way.  I can drop this now if you think it would be better not to have this around.